### DISCUSSION TITLE
Hardware advice ESP8266 NodeMCU V3

### DISCUSSION THREAD

**Discussion by bobblkabb** [micropython/micropython#14369] [Hardware & Peripherals]
Hello,

I would like to control several **servomotors** (approx. 15 motors) with a **NodeMCU v3**. These servomotors are to move small, perhaps 2-3cm large plastic elements.

Which components are recommended for this? How can I control several motors with one board, what do I need for this?
Which servo motors? As small as possible and low voltage would be practical.

Regards,
Michael

**Comment by scruss** [comment]
A [PCA9685](https://www.adafruit.com/product/815) board can drive up to 16 servos via a single I2C connection. There are libraries for MicroPython that support it: [Servo](https://awesome-micropython.com/#servo).

You'll probably need a fairly powerful external supply to power all of those servos.

**Comment by ShaunaB1** [comment]
To control 15 servomotors with a NodeMCU V3, use a motor driver or shield for enough power, connect with jumper wires, and program with the Arduino IDE using the Servo.h library. Choose small, low-voltage micro servos for your project. [Can dog eat](https://www.doglikesbest.com/)

**Reply by scruss** [reply]
This is the **MicroPython** discussion, though ...
